cli/commands.go

```diff
@@ -137,7 +137,7 @@ func CreateCluster(c *cli.Context) error {
 		for i := 0; i < c.Int("workers"); i++ {
 			workerID, err := createWorker(
 				c.GlobalBool("verbose"),
-				fmt.Sprintf("docker.io/rancher/k3s:%s", c.String("version")),
+				fmt.Sprintf("%s:%s", c.String("image"), c.String("version")),
```

Comment:
This is certainly the most straight forward way of doing this, but UX wise it leaves a little to be desired. 

if I specify an image `--image myimage` then I also need to specify a version `--version latest` which is a bit ugly. Since, `myimage` should be assumed latest. Given that this is most often used in development, I think it is a common scenario. 

What would be nice is to do: `--image my/image:v1.2.3` to just follow normal image conventions.

Reply:
I agree with @cloudnautique. 

How about we make -i or --image accept:

1) a fully qualified image name, as suggested by @cloudnautique.
2) accept an image tag that starts with ":", eg: ":latest"

With the 2) option, it effectively cover today's --version option, with the benefit of having only single option for specifying k3s image.

We can depreciate the --version option in some future release.

WDYT?

Reply:
I think if you set the default image to `rancher/k3s:v<official>`

Start deprecating the `version` flag now with a `[Warn] this flag will be deprecated, please use --image rancher/k3s:<version>` if it is used. 

if `--image` is used then just use the image.

if both `--image` AND `--version` are passed, error out and let the user know to use `--image blah/blah:version`

Reply:
This looks clean to me.  Thanks @cloudnautique

Reply:
I also thought about this before, but I didn't want to break with the existing `--version`/`--tag` flag.
I first thought about making `--image image:tag` and `--image <image> --tag <tag>`.
But I think that your approach is cleaner :+1: 
WDYT @zeerorg ?

Reply:
A single `--image` approach seems cleaner than having both `latest` and `version` tags.

Reply:
Thanks for the healthy discussion here guys!
According to the feedback, I updated the PR in Commit 40fb583 .

Changes:
* set `--version` + `--image` = warning + error
* set `--version` = warning + default image with custom tag (version)
* set `--image` = custom image (+ tag)
* default registry is `docker.io` (since we need to specify the registry if the image needs to be pulled)
* default k3s image is `docker.io/rancher/k3s`

Reply:
Since this fits to what @cloudnautique proposed, I'll resolve this conversation here and merge the PR :+1: